webapp/src/components/jira_label_selector/jira_label_selector.tsx

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import AsyncSelect from 'react-select/async';
+import debounce from 'debounce-promise';
+
+import Setting from 'components/setting';
+import {getStyleForReactSelect} from 'utils/styles';
+import {ReactSelectOption} from 'types/model';
+
+const searchDebounceDelay = 400;
+
+type Props = {
+    required?: boolean;
+    hideRequiredStar?: boolean;
+    theme: object;
+    searchLabels: (params: object) => Promise<any>;
+    onChange: (values: string[]) => void;
+    value: string[];
+    addValidate: (isValid: () => boolean) => void;
+    removeValidate: (isValid: () => boolean) => void;
+    resetInvalidOnChange?: boolean;
+};
+
+type State = {
+    invalid: boolean;
+    error?: string;
+    cachedSelectedOptions: ReactSelectOption[];
+};
+
+export default class JiraLabelSelector extends React.PureComponent<Props, State> {
```

Comment:
@AGMETEOR Do you think there is an opportunity here to have this component be reusable for custom label types, or something like a user search selector? I think the autocomplete endpoints for Jira are very similar.

I'm thinking the frontend can pass to the backend the name of the field it wants to search on, and the backend can construct the url from there.

Reply:
I have implemented this